android/src/main/java/com/swmansion/reanimated/nodes/PropsNode.java

```diff
@@ -89,12 +89,32 @@ protected Double evaluate() {
         }
       } else {
         String key = entry.getKey();
+        Object value = node.value();
+        WritableMap dest;
+
         if (mNodesManager.uiProps.contains(key)) {
           hasUIProps = true;
-          mPropMap.putDouble(key, node.doubleValue());
-        } else {
+          dest = mPropMap;
+        } else if (mNodesManager.nativeProps.contains(key)){
           hasNativeProps = true;
-          nativeProps.putDouble(key, node.doubleValue());
+          dest = nativeProps;
+        } else {
+          hasJSProps = true;
+          dest = jsProps;
+        }
+
+        if (value == null) {
+          dest.putNull(key);
+        } else if (value instanceof Number) {
```

Comment:
Could you please write some abstraction for it? Code 30 lines above seems to be very similar and can't be reuse it?

Reply:
I'll dust off my java and try to figure something but I'm not sure if that will be convenient since I want to allow booleans as well and they are invalid prop 30 lines above.

Reply:
I think you could add boolean. I don't allow them since I haven't found any use case, but afaik we could handle booleans as well